File: pytools/examples/golang/src/server/proto/ack_role_login_ok.go
package proto

import (
	"packet"
)

type AckRoleLoginOk struct {
	uname                    string
}

func (this *AckRoleLoginOk) Encode() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteString(this.uname)

	return pack.Encode(P_ACK_ROLE_LOGIN_OK)
}

func (this *AckRoleLoginOk) SetUname(uname string) {
	this.uname = uname
}

File: pytools/examples/lua53/server/protocols/AckRoleLoginOk.lua
AckRoleLoginOk = {}
setmetatable(AckRoleLoginOk, {__index = _G})
local _ENV = AckRoleLoginOk


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = AckRoleLoginOk})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.uname = 0
end


function encode(self)
	pack = packet.new()
	pack:writeString(self.uname)
	return pack:encode(Msg.P_ACK_ROLE_LOGIN_OK)
end


function setUname(self, uname)
	self.uname = uname
end

